osc-operator: remove controller-gen pre-fetch workaround

littlejawa opened the pull request:
"go tool" is a feature introduced with go v1.24 [0], that lets developpers declare a tool that is needed for their project, and manage its dependency along with any other.

This removes the need to make a blank import from our main.go file. We can also remove the script that calls "go run" for downstream builds, as now we can use "go tool" even within the Makefile.

[0] https://go.dev/doc/modules/managing-dependencies#tools

gkurz commented on Makefile:
```diff
@@ -223,7 +223,7 @@ $(LOCALBIN):
 KUBECTL ?= kubectl
 KUSTOMIZE ?= $(LOCALBIN)/kustomize
-CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen
+CONTROLLER_GEN ?= go tool sigs.k8s.io/controller-tools/cmd/controller-gen
```
What about _just_ calling that in https://github.com/openshift/sandboxed-containers-operator/blob/devel/controller-gen#L15 ? This would prevent the Makefile to diverge too much from Operator SDK.

littlejawa replied:
Yeah. I was hoping we could clean things up, but I guess you're right. As long as we use operator-sdk, we need to keep things compatible :-(

gkurz replied:
This still is a great improvement ! :smile:

littlejawa committed:
osc-operator: use 'go tool' to manage controller-gen dependencies

"go tool" is a feature introduced with go v1.24 [0], that lets developpers
declare a tool that is needed for their project, and manage its dependency
along with any other.

This removes the need to make a blank import from our main.go file.

[0] https://go.dev/doc/modules/managing-dependencies#tools

Signed-off-by: Julien Ropé <jrope@redhat.com>

gkurz reviewed (approved):
/lgtm

Thanks @littlejawa !
